### PR TITLE
Added wordpress abilities to bot

### PR DESCRIPTION
**Pull request by d3rpp**

### CODE REVIEW

File: prisma/schema.prisma [hunk 1]
```diff
@@ -1,25 +1,29 @@
...
+}
+
+model LastChecked {
+    last    DateTime
+    guildId String      @id
+    guild   GuildConfig @relation(fields: [guildId], references: [guildId])
+}
+
+model PostedId {
+    id    Int           @id
```
Comment (dylhack) [inline prisma/schema.prisma]: we should add a `wordpress String` (the enable this bot to utilize multiple Wordpress instances at once.

File: src/main.ts [hunk 2]
```diff
@@ -1,9 +1,16 @@
 import startBot from './bot';
 import post from './bot/posts';
-import decodeHe from './wordpress/util';
+import { getUpdates } from './wordpress/update';
+// import decodeHe from './wordpress/util';
```
Comment (dylhack) [inline src/main.ts]: you can delete this line, this was my mistake

File: src/wordpress/index.ts [hunk 3]
```diff
@@ -1,54 +1,50 @@
...
-            reject(e);
-        }
-    });
-};
+import { Article } from '../types/wordpress/Article';
+import { User } from '../types/wordpress/User';
+
+import { ArticlePost } from './article';
+
+export const API_ROOT = 'https://randy.gg';
```
Comment (dylhack) [inline src/wordpress/index.ts]: delete this const

File: src/wordpress/index.ts [hunk 4]
```diff
@@ -1,54 +1,50 @@
...
+  try {
+    // gets Post with id
+    const articleResponse = await fetch(
+      `${APIRoot}/wp-json/wp/v2/posts/${postId}?_fields=${API_QUERY_FIELDS}`,
+    );
+    const articleResponseTyped = (await articleResponse.json()) as Article;
+
+    // Get Author Props
+    const authorResponse = await fetch(
+      `${API_ROOT}/wp-json/wp/v2/users/${articleResponseTyped.author}?_fields=name,avatar_urls`,
```
Comment (dylhack) [inline src/wordpress/index.ts]: replace `API_ROOT` with APIRoot

File: src/wordpress/update.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    const postResponses = await fetch(`${guildConfig.wordpress}/wp-json/wp/v2/posts?_fields=id`);
+    const ids: PostResponse = await postResponses.json();
+
+    const idsToCheck = ids.map((item) => item.id);
+    const idsNotPosted = await checkIdsInDB(idsToCheck);
+
+    idsNotPosted.forEach(async (item) => {
+      const articleDetails = await getArticleDetails(guildConfig.wordpress, item.toString(10));
+
+      callback(guildConfig, articleDetails).then(() => {
```
Comment (dylhack) [inline src/wordpress/update.ts]: perhaps use await since the async/await syntax is already used in this forEach callback?

**Review by dylhack — DISMISSED** [review]

File: prisma/schema.prisma (hunk 1 above)
Comment (dylhack) [reply prisma/schema.prisma:28]: `wordpress String @id` then the id won't be the ID since it wouldn't be unique anymore

Comment (d3rpp) [reply prisma/schema.prisma]: then it'd only do one of the servers since they have the same Wordpress url

Comment (d3rpp) [reply prisma/schema.prisma]: wait no nvm

**Review by dylhack — APPROVED** [review]

**Pull request merged**